dracut/90stratis/stratis-rootfs-setup

```diff
@@ -17,10 +17,14 @@ done
 
 if $(stratis-min pool is-stopped "$STRATIS_ROOTFS_UUID"); then
 	if $(stratis-min pool is-encrypted "$STRATIS_ROOTFS_UUID"); then
-		if ! plymouth ask-for-password \
-			--command="stratis-min pool start --prompt --unlock-method=keyring $STRATIS_ROOTFS_UUID" \
-			--prompt="Enter password for Stratis pool with UUID $STRATIS_ROOTFS_UUID containing root filesystem" \
-			--number-of-tries=3; then
+		ATTEMPTS_REMAINING=3
+		if
+			! while [ $((ATTEMPTS_REMAINING--)) -gt 0 ]
+			do
+				systemd-ask-password --id="stratis:$STRATIS_ROOTFS_UUID" "Enter password for Stratis pool with UUID $STRATIS_ROOTFS_UUID containing root filesystem" |
+				stratis-min pool start --prompt --unlock-method=keyring "$STRATIS_ROOTFS_UUID" && break
+			done
+		then
```

Comment:
In general, I've been unable to find any documentation on what a while loop actually evaluates to in bash. I think this should be restructured to have the while loop first, setting `$RC` to one or zero based on `$?` after the piped command is run, and then the if statement should come after, checking the value of `$RC`.

Reply:
How a while loop evaluates is documented in the bash manual:
https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#index-while
That said, I've made the change you've proposed. It made the code a little longer and `$RC` is checked in two different places, but if it helps you make sense of the code better, then why not…

Reply:
The documentation is sufficient. If you want to drop that last commit, I've tested it and it works. I'm willing to approve just the first two commits.